src/main/java/seedu/address/logic/commands/DeleteEventByIndexCommand.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.Event;
+
+/**
+ * Deletes an event identified using its displayed index from the address book.
+ */
+public class DeleteEventByIndexCommand extends DeleteEventCommand {
+    private Index targetIndex;
+
+    public DeleteEventByIndexCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> lastShownList = model.getFilteredEventList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        }
+
+        Event eventToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        model.deleteEvent(eventToDelete);
+        return new CommandResult(
+                String.format(DeleteEventCommand.MESSAGE_DELETE_EVENT_SUCCESS, Messages.format(eventToDelete)));
```

Comment:
You can use the other `CommandResult` constructor for event type commands:
`public CommandResult(String feedbackToUser, boolean showEventList)`

This constructor helps the app to stay on the event list page, rather than returning to the person list view.

Reply:
Okay! I will change this.